### DISCUSSION TITLE
Auto start shizuku on non-root device (MacroDroid)

### DISCUSSION THREAD

**Discussion by MohabMohie** [RikkaApps/Shizuku#461] [Show and tell]
It's a big topic in the community to auto start Shizuku on a non-root phone.

Here's what I did:
1. Installed MacroDroid.
2. Installed the MacroDroid helper apk. It wasn't supported on my Android 14 Pixel 6, so I installed `PI` which is an application installer that uses Shizuku to bypass some restrictions such as the minimum SDK version.
3. Activated MacroDroid ADB Hack. I used `aShell You` which is also a Shizuku app to execute the ADB commands.
4. Using MacroDroid I created the following job:
![Screenshot_20240329-181218](https://github.com/RikkaApps/Shizuku/assets/19201898/46899308-d41c-45e2-8628-a4af96860fa1)

Here's how it works so that you can replicate it:
- I use the trigger (being connected to my home WiFi)
- I set the `adb_wifi_enabled` system setting to enable WiFi Debugging.
- Then I launch Shizuku itself.
- I insert a synchronization point here to wait until the message `Shizuku is not running` is displayed.
- I click the `Start` button.
- Another synchronization point to wait until the message `Service started` is displayed on screen.
- Lastly I go back to the home screen.

The entire script takes around 2 seconds to execute. And it works everytime I'm connected to the home network.

I may add more constraints to it, such as confirm that the screen is not locked and so on to handle some edge cases where the script will misstrigger. But right now it's working flawlessly when I manually enable the WiFi toggle and my phone gets connected to my home WiFi automatically.

**Comment by chaoscreater** [comment]
Thanks for sharing. Best to upload to the template store and share the macro ID for people to just download easily.

Also, you don't need to keep wireless debugging on in order to keep Shizuku alive. Should toggle it to off at the end.

**Comment by bober757** [comment]


**Reply by SiriosDev** [reply]
cuz (Android/data) is not accessible

**Comment by rahulkushwaha12316-prog** [comment]
Biughjjiiikjjj

**Comment by rahulkushwaha12316-prog** [comment]
Ghefyyu

**Comment by senposage** [comment]
export this macro and upload it somewhere so it can be easily imported

**Comment by ayinghansethan-com** [comment]
shizuko is nat satarting
